index.js

```diff
@@ -26,19 +29,30 @@ module.exports = function (src, dest, opts) {
 
 	opts = objectAssign({overwrite: true}, opts);
 
-	return new Promise(function startRead(resolve, reject) {
-		var read = fs.createReadStream(src);
-
-		read.on('error', function (err) {
-			reject(new CpFileError('cannot read from `' + src + '`: ' + err.message, err));
-		});
+	var progress = {
+		stat: null,
+		emitter: new EventEmitter()
+	};
```

Comment:
As we shouldn't reuse `stat` for updating the access-times etc., we may use only the size:
```diff
- var stat = null;
+ var size = null;
// …
var promise = fsP.stat(src).then(function (stat) {
+		size = stat.size;
```